src/contrib/cluster/Akka.DistributedData.Tests/ReplicatorSpecs.cs

```diff
@@ -0,0 +1,203 @@
+
+            // subscribe to updates for KeyH, then update it with a replication factor of two
+            _replicator2.Tell(Dsl.Subscribe(KeyH, changedProbe.Ref));
+            _replicator2.Tell(Dsl.Update(KeyH, ORDictionary<string, Flag>.Empty, _writeTwo, x => x.SetItem(Cluster.Cluster.Get(_sys2), "a", Flag.False)));
+
+            // receive local update
+            changedProbe.ExpectMsg<Changed>(g => Equals(g.Key, KeyH)).Get(KeyH).Entries.SequenceEqual(ImmutableDictionary.CreateRange(new[]
+            {
+                new KeyValuePair<string, Flag>("a", Flag.False),
+            })).ShouldBeTrue();
+
+            // push update from node 1
+            _replicator1.Tell(Dsl.Update(KeyH, ORDictionary<string, Flag>.Empty, _writeTwo, x => x.SetItem(Cluster.Cluster.Get(_sys1), "a", Flag.True)));
+
+            // expect replication of update on node 2
+            changedProbe.ExpectMsg<Changed>(g => Equals(g.Key, KeyH)).Get(KeyH).Entries.SequenceEqual(ImmutableDictionary.CreateRange(new[]
+            {
+                new KeyValuePair<string, Flag>("a", Flag.True)
+            })).ShouldBeTrue();
+
+            // add new value to dictionary from node 2
+            _replicator2.Tell(Dsl.Update(KeyH, ORDictionary<string, Flag>.Empty, _writeTwo, x => x.SetItem(Cluster.Cluster.Get(_sys2), "b", Flag.True)));
+            changedProbe.ExpectMsg<Changed>(g => Equals(g.Key, KeyH)).Get(KeyH).Entries.SequenceEqual(ImmutableDictionary.CreateRange(new[]
+            {
+                new KeyValuePair<string, Flag>("a", Flag.True),
+                new KeyValuePair<string, Flag>("b", Flag.True)
+            })).ShouldBeTrue();
+        }
+
+        // <summary>
+        /// Reproduction spec for https://github.com/akkadotnet/akka.net/issues/4184
+        /// </summary>
+        [Fact]
+        public async Task Bugfix_Duplicate_Publish()
+        {
+            await InitCluster();
+            await ReplicatorDuplicatePublish();
+        }
+
+        private async Task ReplicatorDuplicatePublish()
+        {
+            var p1 = CreateTestProbe(_sys1);
+            var p2 = CreateTestProbe(_sys2);
+            var p3 = CreateTestProbe(_sys3);
+
+            var probes = new[] { p1, p2, p3 };
+
+            // subscribe to updates on all 3 nodes
+            _replicator1.Tell(Dsl.Subscribe(KeyI, p1.Ref));
+            _replicator2.Tell(Dsl.Subscribe(KeyI, p2.Ref));
+            _replicator3.Tell(Dsl.Subscribe(KeyI, p3.Ref));
+
+            // update item on 2
+            _replicator2.Tell(Dsl.Update(KeyI, GSet<string>.Empty, _writeTwo, a => a.Add("a")));
+
+            // wait for write to replicate to all 3 nodes
+            Within(TimeSpan.FromSeconds(5), () =>
+            {
+                foreach(var p in probes)
+                    p.ExpectMsg<Changed>(c => c.Get(KeyI).Elements.ShouldBe(ImmutableHashSet.Create("a")));
+            });
+
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
Passes when delay is set to 5 seconds, fails with 1 second. This bug must be related to the timing of events and interval-based messages.